rfcs/0000-c-unwind-and-project-group-charter.md

```diff
@@ -0,0 +1,496 @@
+expect the "project group" to largely define its own structure.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+## The "C" ABI and unwinding
+
+Functions declared with the "C" ABI are not permitted to unwind.  In
+general, any attempt to unwind constitutes [Undefined Behavior], which
+means that the program may do arbitrary things.
+
+### Invoking foreign functions
+
+This means, for example, that if you invoke a native "C" function and it
+attempts to unwind (for example, because it invokes C++ code which contains a `throw`),
+this result is [Undefined Behavior]:
+
+```rust
+extern "C" { fn fn_that_is_not_supposed_to_unwind(); }
+
+fn foo() {
+    // If `fn_that_is_not_supposed_to_unwind` should actually unwind, 
+    // the result is undefined behavior.
+    fn_that_is_not_supposed_to_unwind();
+}
+```
+
+### Rust functions with C ABI
+
+In the case of Rust functions declared with the "C" ABI, attempts to
+unwind over the function boundary will result in an abort:
+
+```rust
+extern "C" fn foo() {
+    // This panic will trigger an unwind that tries to pass over an extern "C" 
+    // boundary. It will trigger an abort.
+    panic!();
+}
+```
+
+## The "C unwind" ABI and unwinding
+
+In contrast, functions declared with the "C unwind" ABI **are**
+permitted to unwind. The precise unwinding mechanism in use will
+depend on the target platform, but the intent is to match the
+mechanism used by other systems programming language implementations,
+such as C++.
+
+**WARNING:** The specifiation for the "C unwind" ABI is being actively
+developed and is currently very incomplete. In particular, it is **not
+currently possible to unwind between Rust and non-Rust frames in a
+fully specified fashion**. If you would like to learn more about the
+specification effort, or to get involved, please visit the repository
+for the [ffi-unwind project], which contains a lot more details.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+## Platform dependent behavior of "C" and "C unwind"
+
+In general, the behavior of both the "C" and the "C unwind" ABIs is
+highly dependent on the target triple and other switches that are
+given to the Rust compiler. Therefore, there isn't a lot that can be
+said that is fully independent of the platform.
+
+## Relationship between "C" and "C unwind"
+
+There is no a priori relationship between the "C" and "C unwind" ABIs.
+However, on most platforms, it is expected that the "C" ABI would be
+"compatible" with the "C unwind" ABI. In other words, consider some C
+function `foo` that does not unwind: on most platforms, such a
+function could safely be invoked with either the "C" or the "C unwind"
+ABI. 
+
+However, we do not guarantee such interop: this leaves room for
+platforms where unwinding is transmitted through a special return
+value or other such mechanism. (XXX do any such platforms exist? There
+is the C++ proposal)
```

Comment:
So I'm not sure about these two paragraphs. I suppose that this a defense against "unknown unknowns" but I understand "C unwind" as "the C ABI + unwinding", so I kind of expect it to be equivalent to "C" when `-C panic=abort`, for example. 

The C++ proposal is compatible with the C ABI, and there is also a C proposal for it that adds support for the same C++ feature to C without breaking the current call ABI (I'm not sure if it breaks the unwind ABI on some platforms though, but since the C standard doesn't have a concept of unwinding, it breaks nothing there at least).